### PR TITLE
[BugFix] Fixed orc varchar load failed

### PR DESCRIPTION
**Pull request by zaorangyang**
Fixes #https://github.com/StarRocks/StarRocksTest/issues/2782

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4

### CODE REVIEW

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 1]
```diff
@@ -408,6 +408,9 @@ static void _try_implicit_cast(TypeDescriptor* from, const TypeDescriptor& to) {
         } else {
             from->type = LogicalType::TYPE_DECIMAL32;
         }
+    } else if (_broker_load_mode && !_strict_mode && t1 == LogicalType::TYPE_VARCHAR &&
```
Comment (chaoyli) [inline be/src/formats/orc/orc_chunk_reader.cpp:411]: 1. The issue number is wrong.
2. It's better to add a unit test case to this bug.

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 2]
```diff
@@ -408,6 +408,9 @@ static void _try_implicit_cast(TypeDescriptor* from, const TypeDescriptor& to) {
...
+               t2 == LogicalType::TYPE_VARCHAR) {
```
Comment (Smith-Cruise) [inline be/src/formats/orc/orc_chunk_reader.cpp:412]: add some comment

**Review by Smith-Cruise — APPROVED** [review]

File: be/src/formats/orc/orc_chunk_reader.cpp [hunk 3]
```diff
@@ -408,6 +408,12 @@ static void _try_implicit_cast(TypeDescriptor* from, const TypeDescriptor& to) {
...
+        // For broker load, the orc field length is larger than the maximum length of the starrocks field 
+        // will cause load failure in non-strict mode. Here we keep the maximum length of the orc field 
+        // the same as the maximum length of the starrocks field.
+        from->len = to.len;
```
Comment (rickif) [inline be/src/formats/orc/orc_chunk_reader.cpp:416]: According to the definition of strict mode, it seems that the result should be a NULL.
@jaogoy PTAL.
https://docs.starrocks.io/en-us/2.3/loading/load_concept/strict_mode#:~:text=Strict%20mode%20works%20as%20follows%3A%201%20If%20strict,StarRocks%20loads%20qualified%20rows%20together%20with%20error%20rows.

Comment (zaorangyang) [reply be/src/formats/orc/orc_chunk_reader.cpp:416]: @rickif For non-strict mode， data can be loaded successfully, but overlimited columns should be null.

File: be/test/formats/orc/orc_chunk_reader_test.cpp [hunk 4]
```diff
@@ -1100,6 +1100,50 @@ TEST_F(OrcChunkReaderTest, TestReadBinaryColumn) {
...
+    };
+    slot_descs[1].type.len = 6;
+    slot_descs[2].type.len = 6;
+    static const std::string input_orc_file = "./be/test/exec/test_data/orc_scanner/10380.orc";
+    std::vector<SlotDescriptor*> src_slot_descriptors;
+    ObjectPool pool;
+    create_slot_descriptors(_runtime_state.get(), &pool, &src_slot_descriptors, slot_descs);
+
+    {
+        OrcChunkReader reader(_runtime_state->chunk_size(), src_slot_descriptors);
```
Comment (Smith-Cruise) [inline be/test/formats/orc/orc_chunk_reader_test.cpp:1124]: Test with non-strict mode and strict mode.

File: be/test/formats/orc/orc_chunk_reader_test.cpp [hunk 5]
```diff
@@ -1100,6 +1100,50 @@ TEST_F(OrcChunkReaderTest, TestReadBinaryColumn) {
...
+        reader.set_broker_load_mode(false);
```
Comment (Smith-Cruise) [inline be/test/formats/orc/orc_chunk_reader_test.cpp:1125]: And why disable broker load mode

Comment (zaorangyang) [reply be/test/formats/orc/orc_chunk_reader_test.cpp:1125]: @Smith-Cruise set_broker_load_mode(false) indicates that the strict mode is disabled.

**Review by Smith-Cruise — APPROVED** [review]

**Review by chaoyli — DISMISSED** [review]

**Review by chaoyli — APPROVED** [review]

**Review by meegoo — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**